workers/unity/Packages/io.improbable.gdk.loadbalancing/Systems/SingletonStrategySystem.cs

```diff
@@ -0,0 +1,95 @@
+using System.Linq;
+using Improbable.Gdk.Core;
+using Unity.Collections;
+using Unity.Entities;
+
+namespace Improbable.Gdk.LoadBalancing
+{
+    [UpdateInGroup(typeof(SpatialOSUpdateGroup))]
+    [DisableAutoCreation]
+    [AlwaysUpdateSystem]
+    internal class SingletonStrategySystem : ComponentSystem
+    {
+        private readonly EntityLoadBalancingMap entityLoadBalancingMap;
+        private readonly string workerType;
+
+        private ILogDispatcher logger;
+        private EntityQuery unassignedEntities;
+        private EntityQuery workers;
+
+        private RegisteredWorker targetWorker;
+
+        public SingletonStrategySystem(string workerType, EntityLoadBalancingMap entityLoadBalancingMap)
+        {
+            this.entityLoadBalancingMap = entityLoadBalancingMap;
+            this.workerType = workerType;
+            targetWorker = default;
+        }
+
+        protected override void OnCreate()
+        {
+            logger = World.GetExistingSystem<WorkerSystem>().LogDispatcher;
+
+            unassignedEntities = GetEntityQuery(
+                ComponentType.ReadOnly<Metadata.Component>(),
+                ComponentType.ReadWrite<AuthorityDelegation.Component>(),
+                ComponentType.Exclude<LoadBalancedEntity>());
+
+            workers = GetEntityQuery(ComponentType.ReadOnly<RegisteredWorker>(),
+                ComponentType.ReadOnly<WorkerClassification>());
+            workers.SetSharedComponentFilter(new WorkerClassification(workerType));
+        }
+
+        protected override void OnUpdate()
+        {
+            FindTargetWorker();
+            UpdateEntities();
+        }
+
+        private void FindTargetWorker()
+        {
+            var workerRegistrations = workers.ToComponentDataArray<RegisteredWorker>(Allocator.Temp);
```

Comment:
You should be able to use the `GetSingleton` api for this

Reply:
There's no guarantee there's only one worker connected?

Reply:
True